orttraining/orttraining/python/training/training_agent.py

```diff
@@ -0,0 +1,85 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import collections
+import collections.abc
+import os
+
+import onnxruntime
+from onnxruntime.capi import _pybind_state as C
+# TODO: This create a circular dependency
+from onnxruntime.capi.onnxruntime_inference_collection import IOBinding, OrtValue
+from onnxruntime.capi._pybind_state import TrainingAgent as C_TrainingAgent
+
+
+class TrainingAgent(object):
+    """
+    This is the main class used to run a ORTModule model.
+    """
+
+    def __init__(self, path_or_bytes, session_options=None, providers=None, provider_options=None):
+        """
+        :param path_or_bytes: filename or serialized ONNX or ORT format model in a byte string
+        :param sess_options: session options
+        :param providers: Optional sequence of providers in order of decreasing
+            precedence. Values can either be provider names or tuples of
+            (provider name, options dict). If not provided, then all available
+            providers are used with the default precedence.
+        :param provider_options: Optional sequence of options dicts corresponding
+            to the providers listed in 'providers'.
+
+        The model type will be inferred unless explicitly set in the SessionOptions.
+        To explicitly set:
+          so = onnxruntime.SessionOptions()
+          so.add_session_config_entry('session.load_model_format', 'ONNX') or
+          so.add_session_config_entry('session.load_model_format', 'ORT') or
+
+        A file extension of '.ort' will be inferred as an ORT format model.
+        All other filenames are assumed to be ONNX format models.
+
+        'providers' can contain either names or names and options. When any options
+        are given in 'providers', 'provider_options' should not be used.
+
+        The list of providers is ordered by precedence. For example ['CUDAExecutionProvider', 'CPUExecutionProvider']
+        means execute a node using CUDAExecutionProvider if capable, otherwise execute using CPUExecutionProvider.
+        """
+
+        self._training_agent = None
+        self._inference_session = None
+
+        self._session_options = session_options
```

Comment:
Applying Occam's Razor here. 
self._session_options, self._providers, self._provider_options are not needed for now, we can add them latter if we see need.